install/kubernetes/cilium/values.yaml.tmpl

```
@@ -812,6 +812,8 @@ gatewayAPI:
   enableProxyProtocol: false
   # -- Enable Backend Protocol selection support (GEP-1911) for Gateway API via appProtocol.
   enableAppProtocol: false
+  # -- Enable ALPN for Gateway API.
```

```suggestion
  # -- Enable ALPN for all listeners configured with Gateway API. ALPN will attempt HTTP/2, then HTTP 1.1. Note that this will also enable `appProtocol` support, and services that wish to use HTTP/2 will need to indicate that via their `appProtocol`.
```
Maybe just to make this one hundred percent clear? Ideally, we would add a tutorial to `Documentation/network/servicemesh/gateway-api/` as well walking through what's required.

---

Can we leave the documentation part for a different PR? I would like to actually run this in my env before writing any tutorials. 😅 

Will add the detailed description for the helm values.

---

Circling back on this, as a user it is fairly unclear which direction the ALPN is in, i.e, client <-> envoy or envoy <-> backend, or both. "listeners" implies the former to me, but "appProtocol" implies the latter. Could we write that tutorial now? 😅

---

You're right, alpn is for client <-> envoy and appProtocol is for envoy <-> backend. I'll try to free some cycles for this.